stubs/tensorflow/tensorflow/keras/activations.pyi

```diff
@@ -1,12 +1,37 @@
-from _typeshed import Incomplete
 from collections.abc import Callable
 from typing import Any
 from typing_extensions import TypeAlias
 
+import numpy as np
 from tensorflow import Tensor
+from tensorflow._aliases import FloatArray, FloatDataSequence, FloatTensorCompatible, Integer
 
 # The implementation uses isinstance so it must be dict and not any Mapping.
 _Activation: TypeAlias = str | None | Callable[[Tensor], Tensor] | dict[str, Any]
+# Ints are not allowed.
+_ActivationInput: TypeAlias = Tensor | FloatDataSequence | FloatArray | np.number[Any] | float
 
+def deserialize(
+    name: str, custom_objects: dict[str, Callable[..., Any]] | None = None, use_legacy_format: bool = False
+) -> Callable[..., Any]: ...
+def elu(x: _ActivationInput, alpha: FloatTensorCompatible | FloatDataSequence = 1.0) -> Tensor: ...
+def exponential(x: _ActivationInput) -> Tensor: ...
+def gelu(x: _ActivationInput, approximate: bool = False) -> Tensor: ...
 def get(identifier: _Activation) -> Callable[[Tensor], Tensor]: ...
-def __getattr__(name: str) -> Incomplete: ...
+def hard_sigmoid(x: _ActivationInput) -> Tensor: ...
+def linear(x: _ActivationInput) -> Tensor: ...
+def mish(x: _ActivationInput) -> Tensor: ...
+def relu(
+    x: _ActivationInput,
+    alpha: FloatTensorCompatible = 0.0,
+    max_value: FloatTensorCompatible | FloatDataSequence | None = None,
+    threshold: FloatTensorCompatible | FloatDataSequence = 0.0,
+) -> Tensor: ...
+def selu(x: _ActivationInput) -> Tensor: ...
+def serialize(activation: Callable[..., Any], use_legacy_format: bool = False) -> str: ...
```

Comment:
It can also return a dict:

```
In [15]: tensorflow.keras.activations.serialize(len)
Out[15]: 
{'module': 'builtins',
 'class_name': 'builtin_function_or_method',
 'config': 'len',
 'registered_name': 'builtin_function_or_method'}

In [16]: tensorflow.keras.activations.serialize(lambda: 42)
/Users/jelle/py/venvs/py311/lib/python3.11/site-packages/keras/src/activations.py:549: UserWarning: The object being serialized includes a `lambda`. This is unsafe. In order to reload the object, you will have to pass `safe_mode=False` to the loading function. Please avoid using `lambda` in the future, and use named Python functions instead. This is the `lambda` being serialized: tensorflow.keras.activations.serialize(lambda: 42)

  fn_config = serialization_lib.serialize_keras_object(activation)
Out[16]: 
{'value': ('4wAAAAAAAAAAAAAAAAEAAAADAAAA8wYAAACXAGQBUwApAk7pKgAAAKkAcgMAAADzAAAAAPofPGlw\neXRob24taW5wdXQtMTYtYjFiYzU1YjgzOGViPvoIPGxhbWJkYT5yBgAAAAEAAABzBgAAAIAAqHKA\nAHIEAAAA\n',
  None,
  None)}

In [17]: tensorflow.keras.activations.serialize(os.path.exists)
Out[17]: 
{'module': 'builtins',
 'class_name': 'function',
 'config': 'exists',
 'registered_name': 'function'}
```

Reply:
@JelleZijlstra Thanks for pointing it out, I'll fix that. Do you have a method / tool to find this kind of errors ?
For this particular function, looking at the source code would have made it clear that the return type could be a dict. Is it how you found out about it ?

When doing the TensorFlow stubs, I usually try a few inputs, but I can't test everything, so I often rely on what the docs say :disappointed:

Reply:
Just by trying it out. I didn't do that for all functions, I think at first I was mostly curious what this function would do with a builtin.